### PR TITLE
Add provider options for aws call

### PR DESCRIPTION
**Pull request by ziye-dm**
### Proposed changes

Add provider options to set the region right.

### Related issues (optional)

https://github.com/pulumi/pulumi-eks/issues/844

### CODE REVIEW

**Comment by ziye-dm** [comment]
> Please view the PR build - https://github.com/pulumi/pulumi-eks/actions/runs/3875135795
```
$ /home/runner/work/pulumi-eks/pulumi-eks/nodejs/eks/node_modules/.bin/tsc
Error: cluster.ts(921,49): error TS2[55](https://github.com/pulumi/pulumi-eks/actions/runs/3875135795/jobs/6607317577#step:14:56)4: Expected 3 arguments, but got 2.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [Makefile:33: build_nodejs] Error 1
Error: Process completed with exit code 2.
```
My bad. Forget to build locally.

**Comment by ziye-dm** [comment]
@thomas11 Please try the test again?

**Comment by ziye-dm** [comment]
> Please view the PR build - https://github.com/pulumi/pulumi-eks/actions/runs/3896854173

Succeeded :D. Anything else I need to do? @thomas11

**Comment by thomas11** [comment]
Hey @ziye-dm, sorry for the delay. Someone else would like to review the PR before we merge it.

**Comment by ziye-dm** [comment]
@justinvp Friendly ping?

File: nodejs/eks/nodegroup.ts [hunk 1]
```diff
@@ -1197,7 +1197,7 @@ ${customUserData}
                     values: [id],
                 },
             ],
-        }),
+        }, { parent, provider}),
```
Comment (justinvp) [inline nodejs/eks/nodegroup.ts:1200]: I suspect the only change necessary to address the issue is this line, which should be:

```suggestion
        }, { parent }),
```

I suspect all the other changes in this PR to take a `provider` are unnecessary (sorry to lead you astray about `provider`). The provider from the parent should be used implicitly, and it looks like in this one place we missed passing `parent` to the `getAmi` call.

@ziye-dm, would you mind giving that a try and letting us know if it addresses the issue for you? Thanks!

**Review by justinvp — CHANGES_REQUESTED** [review]

File: nodejs/eks/nodegroup.ts (hunk 1 above)
Comment (ziye-dm) [reply nodejs/eks/nodegroup.ts:1200]: Yes only one line change could fix my problem and I have tried it's working. See https://github.com/pulumi/pulumi-eks/issues/844#issuecomment-1373357823

I don't really understand how other places work and could you please answer the question that @thomas11 mentioned?
https://github.com/pulumi/pulumi-eks/issues/844#issuecomment-1373781127
> I still wonder about this line (1058). Shouldn't we add `provider there, too?
> const awsRegion = pulumi.output(aws.getRegion({}, { parent, async: true }));

I am ok to change back to one line change if you two are aligned. :D

Comment (ziye-dm) [reply nodejs/eks/nodegroup.ts]: Get it! Only `parent` is needed! It's working!

**Review by thomas11 — APPROVED** [review]

**Comment by ziye-dm** [comment]
@justinvp  Friendly ping?

**Review by justinvp — APPROVED** [review]

**Pull request merged**